File: solutions/go/0002-add-two-numbers.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	var carry int
	list := &ListNode{}
	last := list

	for l1 != nil || l2 != nil || carry > 0 {
		var num1, num2 int

		if l1 != nil {
			num1 = l1.Val
			l1 = l1.Next
		}

		if l2 != nil {
			num2 = l2.Val
			l2 = l2.Next
		}

		sum := num1 + num2 + carry
		carry = 0

		if sum >= 10 {
			sum -= 10
			carry = 1
		}

		last.Next = &ListNode{Val: sum}
		last = last.Next
	}

	return list.Next
}

File: solutions/php/0002-add-two-numbers.php
<?php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution
{
    /**
     * @param ListNode $l1
     * @param ListNode $l2
     * @return ListNode
     */
    function addTwoNumbers(ListNode $l1, ListNode $l2): ListNode
    {
        $list = new ListNode();
        $last = $list;
        $carry = 0;

        while ($l1 || $l2 || $carry) {
            $sum = ($l1 ? $l1->val : 0) + ($l2 ? $l2->val : 0) + $carry;
            $carry = floor($sum / 10);
            $sum %= 10;

            $last->next = new ListNode($sum);
            $last = $last->next;

            $l1 = $l1?->next;
            $l2 = $l2?->next;
        }

        return $list->next;
    }
}
